### PR TITLE
111 expose astra authentication

### PR DESCRIPTION
**Pull request by alexeiakimov**
The present PR provides

1. Updates to the existing code to be compatible with the cassandra-sys-rs after the later was upgraded to support C++ driver version 2.16.0
2. New API for setting the cloud secure connection bundle
3. A simple example of how to use cloud secure connection bundles

At the moment the PR is a draft, the following items are pending
1. Migration to the new new release of cassandra-sys-rs
2. Updating the changelog and other documents.

@kw217 Could you please review. Thank you.

### CODE REVIEW

File: src/cassandra/error.rs [hunk 1]
```diff
@@ -215,6 +215,7 @@ pub enum CassErrorCode {
     LIB_INVALID_STATE,
     LIB_NO_CUSTOM_PAYLOAD,
     LIB_EXECUTION_PROFILE_INVALID,
```
Comment (kw217) [inline src/cassandra/error.rs:217]: As discussed, please can you add `#[non_exhaustive]` to this enum?

**Review by kw217 — CHANGES_REQUESTED** [review]
Looking good so far - thanks! Please make sure you mention the breaking change in the changelog; I don't think it's important enough to call out in the README with a migration guide though.
Please can you fix the broken build? It looks like it needs to pick up a newer version of the Cassandra driver - take a look at what I did on cassandra-sys https://github.com/Metaswitch/cassandra-sys-rs/pull/40 to see what needs to be done.

**Review by kw217 — CHANGES_REQUESTED** [review]
Thanks @alexeiakimov !
I've now released cassandra-sys-rs 1.0.0; please can you pick that up, and also rebase onto the current master here? Then I think we will be good to go!

**Comment by alexeiakimov** [comment]
> Thanks @alexeiakimov ! I've now released cassandra-sys-rs 1.0.0; please can you pick that up, and also rebase onto the current master here? Then I think we will be good to go!

Thank you. Done.

**Review by kw217 — APPROVED** [review]
LGTM! Thanks @alexeiakimov !

**Comment by kw217** [comment]
I'm keen to do a new release for this, but I have a few other things to check first. Do you have an urgent need for this, or are you OK to wait for a week or so?

**Comment by alexeiakimov** [comment]
@kw217 Thank you so much. No it is not urgent, please take you time.

**Pull request merged**